Core/Core/Assignments/Assignment.swift

```diff
@@ -183,6 +184,21 @@ extension Assignment {
                 self.submissions = nil
             }
         }
+        
+        if updateScoreStatistics {
+            if let newStatistics = item.score_statistics {
+                if let scoreStatistics = scoreStatistics {
+                    scoreStatistics.update(fromApiModel: newStatistics, in: client)
+                } else {
+                    let saveStats = ScoreStatistics(context: client)
+                    saveStats.update(fromApiModel: newStatistics, in: client)
+                    scoreStatistics = saveStats
+                }
```

Comment:
This `if/else` could be simplified to:

```swift
scoreStatistics = scoreStatistics ?? client.insert()
scoreStatistics.update(fromApiModel: newStatistics, in: client)
```

Reply:
Turns out this doesn't actually work: because scoreStatistics is an optional, when insert() tries to convert the type to a string, it gets "Optional<ScoreStatistics>" which is not a valid name of a model. I will still refactor it a bit.